server/sonar-search/src/main/java/org/sonar/search/SearchServer.java

```diff
@@ -50,56 +57,177 @@ public SearchServer(Props props) {
+    }
+
+    String grandParentAssembly = "../" + parentAssembly;
+    Path buildServerPath2 = Paths.get(grandParentAssembly + executable);
+    if (buildServerPath2.toFile().exists()) {
+      return buildServerPath2.toAbsolutePath();
+    }
+
+    throw new IllegalStateException(String.format("Cannot find elasticsearch binary %s in either %s or %s or %s or %s",
+            executable,
+            Paths.get(".").toAbsolutePath().getParent(),
+            Paths.get(assembly).toAbsolutePath(),
+            Paths.get(parentAssembly).toAbsolutePath(),
+            Paths.get(grandParentAssembly).toAbsolutePath()
+    ));
+  }
+
+  private static String getExecutable() {
+    if (System.getProperty("os.name").startsWith("Windows")) {
+      return "elasticsearch/bin/elasticsearch.bat";
+    }
+    return "elasticsearch/bin/elasticsearch";
   }
 
   @Override
   public Status getStatus() {
-    boolean esStatus = node != null && node.client().admin().cluster().prepareHealth()
-        .setWaitForYellowStatus()
-        .setTimeout(TimeValue.timeValueSeconds(30L))
-        .get()
-        .getStatus() != ClusterHealthStatus.RED;
-    if (esStatus) {
-      return Status.OPERATIONAL;
+    Status status = null;
+    try {
+      status = getStatus2();
+      System.out.println("ES STATUS "+status);
+      return status;
+    } catch (Exception e) {
+      System.out.println("ES STATUS "+e.getMessage());
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Status getStatus2() {
+    String urlString = url+"/_cluster/health";
+    try {
+      URL url = new URL(urlString);
+      URLConnection urlConnection = url.openConnection();
+      InputStream inputStream = urlConnection.getInputStream();
+      String line;
+      BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+      while ((line = reader.readLine()) != null) {
+        if (line.contains("\"status\"")) {
+          if (line.contains("\"red\"")) {
+            return Status.DOWN;
+          }
+          if (line.contains("\"yellow\"")) {
+            return Status.OPERATIONAL;
+          }
+          if (line.contains("\"green\"")) {
+            return Status.OPERATIONAL;
+          }
+        }
+      }
+    } catch (MalformedURLException e) {
+      throw new RuntimeException(String.format("Cannot contact local Elasticsearch instance via url '%s'", urlString), e);
+    } catch (IOException e) {
+      e.printStackTrace();
     }
     return Status.DOWN;
   }
 
   @Override
   public void awaitStop() {
     try {
-      while (node != null && !node.isClosed()) {
+      while (p != null && p.isAlive()) {
         Thread.sleep(200L);
       }
     } catch (InterruptedException e) {
-      // Restore the interrupted status
-      Thread.currentThread().interrupt();
+      e.printStackTrace();
+    }
+
+    if (p != null) {
+      p.destroy();
+      try {
+        p.waitFor();
+      } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2142)